lib/bundix/prefetcher.rb

```diff
@@ -69,7 +68,15 @@ def prefetch(source)
     case source
     when Bundix::Source::Gem
       sha = source.rubygems_org? && wrapper.gem(source.name, source.version)
-      sha ||= wrapper.url(source.urls.first) # XXX
+      return sha if sha
+
+      source.urls.find do |url|
+        begin
+          wrapper.url(url)
+        rescue Thor::Error
```

Comment:
Now that `Wrapper#exec` isn't throwing an exception from `thor`, this `rescue` shouldn't use `Thore::Error`, right?